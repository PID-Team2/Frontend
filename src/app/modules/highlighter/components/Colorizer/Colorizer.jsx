function Colorizer({languageElement}) {

  const { word,clasification, error} = languageElement;

  return (
    <pre className={"hlg_"+clasification}>{error ? "Error!" : word}</pre>
  )
}
export default Colorizer