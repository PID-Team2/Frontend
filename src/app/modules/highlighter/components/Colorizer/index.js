export { default as Colorizer } from './Colorizer';
