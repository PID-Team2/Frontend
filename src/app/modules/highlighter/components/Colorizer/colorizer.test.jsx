import {render } from '@testing-library/react';
import Colorizer from './colorizer';

const renderTree = tree => render(tree);
describe('<Colorizer>', () => {
  it('should render component', () => {
    const {asFragment} =  renderTree(<Colorizer languageElement={
      {
      word : 'const',
      clasification : 'declarations',
      error : false
    }}/>)

    expect(asFragment()).toMatchSnapshot()
  });
  
});