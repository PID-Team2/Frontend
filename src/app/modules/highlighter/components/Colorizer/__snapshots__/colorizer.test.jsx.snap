// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Colorizer> should render component 1`] = `
<DocumentFragment>
  <pre
    class="hlg_declarations"
  >
    const
  </pre>
</DocumentFragment>
`;
